src/atoms/Checkbox/Checkbox.web.js

```diff
@@ -1,3 +1,282 @@
+  },
+  backdrop: {
+    backgroundColor({ theme, state, isChecked, externalChecked, disabled, variantColor }) {
+      if (disabled) {
+        return 'transparent';
+      }
+
+      switch (state) {
+        case 'hover':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[930];
+          }
+          return getColor(theme, `${variantColor}.920`);
+        case 'focus':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.930`);
+        case 'active':
+          if (!(isChecked || externalChecked)) {
+            return theme.colors.tone[940];
+          }
+          return getColor(theme, `${variantColor}.940`);
+        default:
+          return 'transparent';
+      }
+    },
+    width: {
+      large: makePxValue(3.5),
+      medium: makePxValue(3),
+      small: makePxValue(2.5),
+      xsmall: makePxValue(2),
+    },
+    height: {
+      large: makePxValue(3.5),
+      medium: makePxValue(3),
+      small: makePxValue(2.5),
+      xsmall: makePxValue(2),
+    },
+    borderRadius: {
+      large: makePxValue(1.75),
+      medium: makePxValue(1.5),
+      small: makePxValue(1.25),
+      xsmall: makePxValue(1),
+    },
+  },
+  title: {
+    color({ disabled }) {
+      if (disabled) {
+        return 'shade.950';
+      }
+      return 'shade.980';
+    },
+  },
+};
+
+const Checkbox = ({
+  defaultChecked,
+  checked: externalChecked,
+  onChange,
+  disabled,
+  title,
+  size,
+  variantColor,
+  helpText,
+  errorText,
+  testID,
+  name,
+  id,
+}) => {
+  if (isDefined(defaultChecked) && isDefined(externalChecked)) {
+    throw Error('One of defaultChecked or checked should be supplied.');
+  }
+
+  const [isChecked, setIsChecked] = useState(defaultChecked || false);
+
+  const onCheckChange = useCallback(
+    (event) => {
+      /* prevent default behaviour of label firing click event on the associated input */
+      event.preventDefault();
+      if (!disabled) {
+        if (isDefined(externalChecked)) {
+          onChange(!externalChecked);
+          return;
+        }
+        setIsChecked((prevState) => {
+          onChange(!prevState);
+          return !prevState;
+        });
+      }
+    },
+    [externalChecked, onChange, disabled],
+  );
+
+  const descriptionText = errorText || helpText;
+
+  return (
+    <Flex alignSelf="flex-start" flexDirection="column">
+      <View>
+        <Flex flexDirection="row" alignItems="center">
+          <Label as="label" onClick={onCheckChange} htmlFor={id}>
+            <Input
+              id={id}
+              name={name}
+              defaultChecked={isChecked || externalChecked}
+              isChecked={isChecked}
+              externalChecked={externalChecked}
+              disabled={disabled}
+              variantColor={variantColor}
+              backdropStyles={styles.backdrop}
+              {...automation(testID)}
```

Comment:
Move this to label.

Reply:
Check if tests are working with this, otherwise move.

Reply:
works